src/shogun/features/DenseFeatures.cpp

```diff
@@ -1036,24 +1036,34 @@ template<class ST> CFeatures* CDenseFeatures<ST>::create_merged_copy(
 	SGMatrix<ST> data(num_features, num_vectors_merged);
 
 	/* copy data of this instance */
-	SG_DEBUG("copying matrix of this instance\n")
-	memcpy(data.matrix, feature_matrix.matrix,
-			num_features*num_vectors*sizeof(ST));
+	SG_DEBUG("copying matrix of this instance\n");
+	if (m_subset_stack->has_subsets())
+	{
+		SGMatrix<ST> active_feats_mat=get_feature_matrix();
```

Comment:
Not a fan of this. Double memory usage.
Either create a method that extract the subset matrix into a pre-allocated one, or copy things by hand here.